File: SOLID/SingleResponsibilityPrinciple/TypeScript/Share.js
"use strict";
exports.__esModule = true;
var Share = /** @class */ (function () {
    function Share(news) {
        this.news = news;
    }
    Share.prototype.magergram = function () {
        var url = 'https://magergram.herokuapp.com';
        return this.news.title + " shared " + url;
    };
    Share.prototype.vk = function () {
        var url = 'https://vk.com';
        return this.news.title + " shared " + url;
    };
    return Share;
}());
exports.Share = Share;

File: SOLID/SingleResponsibilityPrinciple/python/Share.py
from SOLID.SingleResponsibilityPrinciple.python.News import News

class Share:
    news: News

    def __init__(self, news):
        self.news = news

    def magegram(self):
        url = 'https://magergram.herokuapp.com'
        return f"{self.news.title} shared {url}"

    def vk(self):
        url = 'https://vk.com'
        return f"{self.news.title} shared {url}"